modules/ztp-image-based-upgrade-procedure.adoc

```diff
@@ -0,0 +1,144 @@
++
+[source,yaml]
+----
+apiVersion: lcm.openshift.io/v1alpha1
+kind: ImageBasedGroupUpgrade
+metadata:
+  name: example-upgrade
+  namespace: default
+spec:
+  clusterLabelSelectors: <1>
+  - matchExpressions:
+    - key: name
+      operator: In
+      values:
+      - spoke1
+      - spoke2
+      - spoke6
+  ibuSpec:
+    seedImageRef: <2>
+      image: quay.io/seed/image:4.16.0-rc.1
+      version: 4.16.0-rc.1
+      pullSecretRef:
+        name: "<seed_pull_secret>"
+  extraManifests: <3>
+    - name: example-extra-manifests
+      namespace: openshift-lifecycle-agent
+  oadpContent: <4>
+    - name: oadp-cm
+      namespace: openshift-adp
+  plan: <5>
+  - actions:
+    - Prep
+    rolloutStrategy:
+      maxConcurrency: 2
+      timeout: 2400
+  - actions:
+    - Upgrade
+    rolloutStrategy:
+      maxConcurrency: 2
+      timeout: 30
+  - actions:
+    - FinalizeUpgrade
+    rolloutStrategy:
+      maxConcurrency: 10
+      timeout: 3
+----
+<1> Specify the set of clusters that you want to upgrade.
+<2> Defines the target platform version, the seed image to be used, and the secret required to access the image.
+<3> (Optional) Specify the list of `ConfigMap` resources that contain your custom catalog sources to retain after the upgrade, and your extra manifests to apply to the target cluster that are not part of the seed image.
+<4> Specify the list of `ConfigMap` resources that contain the OADP `Backup` and `Restore` CRs.
+<5> Defines the upgrade plan. This example plan shows that all steps have separate rollout strategies.
+// Do we want to include all this? Do we need to add more information?
+
+. Apply the `example-upgrade.yaml` file by running the following command on the hub cluster:
++
+[source,terminal]
+----
+$ oc apply -f example-upgrade.yaml
+----
+
+. Monitor the status updates by running the following command:
++
+--
+[source,terminal]
+----
+$ oc get ibgu -o yaml
+----
+
+[source,yaml]
+----
+[...]
+status:
+  clusters:
+  - completedActions:
+    - action: Prep
+    name: spoke1
+  - completedActions:
+    - action: Prep
+    name: spoke2
+  - failedActions:
+    - action: Prep
+    name: spoke6
+  conditions:
+  - message: All plan steps are completed
+    reason: Completed
+    status: "False"
+    type: Progressing
+[...]
+----
+
+Based on the results of the step, {cgu-operator} adds a label to the clusters.
+In this example, `lcm.openshift.io/ibgu-prep-failed` label is applied to `spoke6` after a failed `Prep` stage transition.
+
+After investigating the reason for failure, you can append an `AbortOnFailure` step to move the cluster back to the `Idle` stage.
+This deletes any resources that are related to the upgrade and allows you to retry the upgrade after troubleshooting.
+--
+
+. (Optional) Append the `AbortOnFailure` action as a separate step to your plan in the `ImageBasedGroupUpgrade` CR:
++
+--
+// Please let me know how you would append this step in practice.
```

Comment:
For appending upgrade to prep for example, create a file with the new contents of plan
```
spec:
  plan:
    - actions: ["Prep"]
      rolloutStrategy:
        maxConcurrency: 200
        timeout: 1
    - actions: ["Upgrade"]
      rolloutStrategy:
        maxConcurrency: 200
        timeout: 30
```
kubectl patch ibgu <ibgu name> --type=merge --patch-file <file path>

you can also use `oc edit ibgu <ibgu name>` or modify the original file and run `oc apply -f ibgu.yaml`